xyz-to-coords.py

```diff
@@ -0,0 +1,38 @@
+import os
+
+def give_coords(filename):
+    directory = r'/home/jacob/scf_guess/geometries'
+    for filename in os.listdir(directory):
+        if filename.endswith(".xyz"):
+            with open(filename) as f:
+                num_atoms = 0
+                counter = 0
+                coords = []
+                for line in f:
+                    counter += 1
+                    if counter == 0:
+                        num_atoms = int(line.split()[0])
+                    elif counter > 1:
+                        element = line.split()
+                        symbol = element[0]
+                        x,y,z = float(element[1]), float(element[2]), float(element[3])
+                        coords.append([symbol, x, y, z])
+                    elif counter == 2+num_atoms:
+                        break
+        if filename.endswith(".sdf"):
+            with open(filename) as f:
+                counter = 0
+                natoms = 0
+                coords = []
+                for line in f:
+                    if counter == 3:
+                        natoms = int(line.split()[0])
+                    elif counter == 4 + natoms:
+                        break
+                    elif counter > 3:
+                        atom = line.split()
+                        at_sym = atom[3]
+                        x,y,z = float(atom[0]), float(atom[1]), float(atom[2])
+                        coords.append([at_sym, x, y, z])
+                    counter += 1
+    return coords
```

Comment:
I tried to test this with "xyz-to-coords("Dimer")" but for some reason it tries to pull "EthaneA.sdf." If this is a cache, I'm not sure to delete it. But with our previous as precedent, this looks ok to me.

Reply:
What have you tried as far as debugging goes? While I'm pretty sure I see your problem, you need to learn how to debug on your own. I'd say over half of writing code is debugging it. When getting started with coding it's often easiest to put a bunch of print statements in the code, run it, and see where your logic goes wrong. If that doesn't help I recommend breaking your function up into smaller functions (say one function which does the xyz and one which does the sdf) and making sure each function works individually before trying to ensure they work together.

Reply:
My apologies, I shouldn't have pushed this without attempting to debug first.

If I print lines 3-6 of the function, I get every file in the directory that ends with ".xyz" as the printed output. I fixed that, but then on line 7 (using "open(filename)") there is an error, stating that file does not exist. Before I add this line, the function can indeed find the inputted file within the directory. What do you suggest? I haven't been able to fix this.

Reply:
You have to figure out why `filename` doesn't exist. Is it a typo? Are you in the wrong directory?